### PR TITLE
added tests for CatNinja

### PR DESCRIPTION
**Pull request by AndreeaMonica**

### CODE REVIEW

**Review by AlexSpinu — CHANGES_REQUESTED** [review]
Fa i te rog si merge cu main-ul actual

File: TestFrame/Tests/CatsTests/CatFactsTests.cs [hunk 1]
```diff
@@ -15,20 +17,55 @@ public class CatFactsTests : OrderedAcceptanceTestsBase<CatsTestsFixture>
...
+            var response = await restFactory.Create()
+                                            .WithRequest("/breeds", Method.Get)
+                                            .WithQueryParameter("limit", $"{TestFixture.Limit}")
+                                            .Execute<GetCatsModel>(TestFixture.Client);
+            var getResponse = response.Data;
 
+            #region Assertions
+            using (new AssertionScope())
+            {
+                getResponse.Data.ElementAt(0).Breed.Should().Be("Abyssinian");
```
Comment (miaunescu) [inline TestFrame/Tests/CatsTests/CatFactsTests.cs:46]: please check for nulls at different levels, Data, Elements.

File: TestFrame/Tests/CatsTests/CatFactsTests.cs [hunk 2]
```diff
@@ -38,12 +75,121 @@ public async Task Get_Cats_Test()
             #region Assertions
             using (new AssertionScope())
             {
-                getResponse.Data.ElementAt(0).Breed.Should().Be("Abyssinian");
```
Comment (miaunescu) [inline TestFrame/Tests/CatsTests/CatFactsTests.cs:41]: same here

File: TestFrame/Tests/CatsTests/CatFactsTests.cs [hunk 3]
```diff
@@ -38,12 +75,121 @@ public async Task Get_Cats_Test()
...
+                                            .WithQueryParameter("max_length", $"{300}")
+                                            .WithQueryParameter("limit", $"{10}")
+                                            .Execute<GetCatsModel>(TestFixture.Client);
+            var getResponse = response.Data;
+
+
+            #region Assertions
+            using (new AssertionScope())
+            {
+                getResponse.PerPage.Should().Be($"{10}");
```
Comment (miaunescu) [inline TestFrame/Tests/CatsTests/CatFactsTests.cs:201]: GetResponse could still be null here

**Review by miaunescu — CHANGES_REQUESTED** [review]

**Review by miaunescu — APPROVED** [review]

**Pull request merged**